### ISSUE TITLE
Jupyter notebook kernel timeout when distributed installed

### ISSUE THREAD

**Issue by tlambert03** [dask/dask#5258]
I'm having what appears to be the same issue as #4897, in which installing distributed gives me `Timeout waiting for kernel_info` in Jupyter notebook/labs.  However, installing a fresh environment, which seems to have worked there, does not work for me.

I'm trying to set this up on my compute cluster running Centos 7 with slurm.  If I start with a clean environment _without_ distributed (`conda create -n dask-test python=3.7 ipython notebook`), then the kernel loads fine when I start notebook (`jupyter notebook --no-browser`).  here's what my environment looks like at that point:
```
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main
attrs                     19.1.0                     py_0    conda-forge
backcall                  0.1.0                      py_0    conda-forge
bleach                    3.1.0                      py_0    conda-forge
bzip2                     1.0.8                h516909a_0    conda-forge
ca-certificates           2019.6.16            hecc5488_0    conda-forge
certifi                   2019.6.16                py37_1    conda-forge
decorator                 4.4.0                      py_0    conda-forge
defusedxml                0.5.0                      py_1    conda-forge
entrypoints               0.3                   py37_1000    conda-forge
ipykernel                 5.1.2            py37h5ca1d4c_0    conda-forge
ipython                   7.7.0            py37h5ca1d4c_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
jedi                      0.14.1                   py37_0    conda-forge
jinja2                    2.10.1                     py_0    conda-forge
jsonschema                3.0.2                    py37_0    conda-forge
jupyter_client            5.3.1                      py_0    conda-forge
jupyter_core              4.4.0                      py_0    conda-forge
libffi                    3.2.1             he1b5a44_1006    conda-forge
libgcc-ng                 9.1.0                hdf63c60_0
libsodium                 1.0.17               h516909a_0    conda-forge
libstdcxx-ng              9.1.0                hdf63c60_0
markupsafe                1.1.1            py37h14c3975_0    conda-forge
mistune                   0.8.4           py37h14c3975_1000    conda-forge
nbconvert                 5.6.0                      py_0    conda-forge
nbformat                  4.4.0                      py_1    conda-forge
ncurses                   6.1               hf484d3e_1002    conda-forge
notebook                  6.0.0                    py37_0    conda-forge
openssl                   1.1.1c               h516909a_0    conda-forge
pandoc                    2.7.3                         0    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
parso                     0.5.1                      py_0    conda-forge
pexpect                   4.7.0                    py37_0    conda-forge
pickleshare               0.7.5                 py37_1000    conda-forge
pip                       19.2.1                   py37_0    conda-forge
prometheus_client         0.7.1                      py_0    conda-forge
prompt_toolkit            2.0.9                      py_0    conda-forge
ptyprocess                0.6.0                   py_1001    conda-forge
pygments                  2.4.2                      py_0    conda-forge
pyrsistent                0.15.4           py37h516909a_0    conda-forge
python                    3.7.3                h33d41f4_1    conda-forge
python-dateutil           2.8.0                      py_0    conda-forge
pyzmq                     18.0.2           py37h1768529_2    conda-forge
readline                  8.0                  hf8c457e_0    conda-forge
send2trash                1.5.0                      py_0    conda-forge
setuptools                41.0.1                   py37_0    conda-forge
six                       1.12.0                py37_1000    conda-forge
sqlite                    3.29.0               hcee41ef_0    conda-forge
terminado                 0.8.2                    py37_0    conda-forge
testpath                  0.4.2                   py_1001    conda-forge
tk                        8.6.9             hed695b0_1002    conda-forge
tornado                   6.0.3            py37h516909a_0    conda-forge
traitlets                 4.3.2                 py37_1000    conda-forge
wcwidth                   0.1.7                      py_1    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.33.4                   py37_0    conda-forge
xz                        5.2.4             h14c3975_1001    conda-forge
zeromq                    4.3.2                he1b5a44_2    conda-forge
zlib                      1.2.11            h516909a_1005    conda-forge
```

If I then `conda install distributed` (distributed  2.2.0,  conda-forge) and restart the kernel in notebook then I get `Timeout waiting for kernel_info_reply` in my logs, notebook doesn't work.  At that point, uninstalling only distributed with `conda uninstall distributed --force` is sufficient for me to be able to restart my kernel in notebook again.  I can go back and forth with `install distributed` and `uninstall distributed --force` and it consistently fails as long as distributed is installed.

after reading through the threads here, I've tried python 3.6 and 3.7, distributed 2.2.0 and 1.28.1 (neither version mattered), and tornado 6.0.3 and 5.1.1 ... only uninstalling distributed gets my kernel to work... but my ultimate goal is to use dask.distributed and dask-jobqueue.

(in case it's important info, notebook is running on a compute node on my cluster and I'm logging in via ssh tunnel through the login node.)

thanks for any suggestions you can provide.

**Comment by mrocklin** [comment]
The only thing that came to mind to me was to check the tornado version,
but it looks like you've already done that.  I personally don't know of any
other possible complication.

**Comment by tlambert03** [comment]
Thanks Matthew. If tornado is at the top of your list, I'll keep playing around with various version combinations.  Was it specifically v6 you were looking for?  or is there reason to believe I might have better luck with v5?

as a side note: I got started on this after coming across your videos on dask w/ jobqueue on HPC, and it has dramatically improved the degree to which I am utilizing my institution's cluster.  So with or without jupyter, thanks so much for all your work.  this is awesome.

**Comment by mrocklin** [comment]
I would not be surprised if some versions of Jupyter do not work with
Tornado 6.  In my recent experience the newest versions of both work well
together though.

**Comment by tlambert03** [comment]
it seems obvious that this has something to do with permissions or the environment on my cluster... installing the exact same conda environment locally works, even with distributed.  I'll ask my admins if they have any ideas.  If anyone has ideas about specific permissions I should have them check (that might cause the timeout loading kernel info) let me know.

**Comment by TomAugspurger** [comment]
@tlambert03 will you report back here with what you find? It can be helpful for future users facing similar issues.

**Comment by tlambert03** [comment]
well, it was my own damn fault (sorry folks)... but I figured it out:  I had some custom imports specified in ~/.ipython/profile_default/ipython_config.py:
```
c.InteractiveShellApp.exec_lines = [
    'import numpy as np',
    'from dask_jobqueue import SLURMCluster',
    'from dask.distributed import Client'
]
```
when distributed _wasn't_ installed, I noticed the following error in my notebook debug logs:
`[D 09:49:06.818 LabApp] activity on c7b01486-a3f2-4157-a41f-78e8fabf31e7: error
`
I edited `notebook.services.kernels.kernelmanager` to provide a bit more feedback on the error and that tipped me off to the import errors when the notebook kernel started.  For some reason, when distributed _was_ installed, those imports didn't throw an error, but did cause the kernel to hang somehow and eventually caused the timeout.  Removing those lines from my ipython config fixed the problem.

Sorry for the issue noise. thanks for your time... and I guess if anyone else stumbles on this, don't forget to check your ipython config at ~/.ipython/profile_default/ipython_config.py
